src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -768,14 +770,35 @@ pub enum StaleItem {
 impl LocalFingerprint {
     /// Read the environment variable of the given env `key`, and creates a new
     /// [`LocalFingerprint::RerunIfEnvChanged`] for it.
-    ///
-    // TODO: This is allowed at this moment. Should figure out if it makes
-    // sense if permitting to read env from the config system.
     #[allow(clippy::disallowed_methods)]
-    fn from_env<K: AsRef<str>>(key: K) -> LocalFingerprint {
-        let key = key.as_ref();
+    fn from_env<K: AsRef<str>>(
```

Comment:
> This would likely need to revert https://github.com/rust-lang/cargo/pull/12482.

Don't we need to update the documentation with this?

Reply:
Sorry for the long delay with work and life stuff.

Does `revert` mean to revert to a previous state?I used google translate.

Reply:
> Sorry for the long delay with work and life stuff.

Fully understand.  I'm behind on a lot of stuff.

> Does revert mean to revert to a previous state?I used google translate.

Yes.  Its a term commonly used in VCS, like git.  Basically "undo".